### ISSUE TITLE
bootkube service not restarting on failure

### ISSUE THREAD

**Issue by abhinavdahiya** [openshift/installer#540]
In one one my ci runs https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/146/pull-ci-openshift-machine-config-operator-master-e2e-aws/117

# Platform (aws|libvirt|openshift):

AWS

# What happened?

```
● bootkube.service - Bootstrap a Kubernetes cluster
   Loaded: loaded (/etc/systemd/system/bootkube.service; static; vendor preset: disabled)
   Active: active (exited) (Result: exit-code) since Thu 2018-10-25 19:34:03 UTC; 1h 33min ago
  Process: 913 ExecStart=/opt/tectonic/bootkube.sh (code=exited, status=125)
 Main PID: 913 (code=exited, status=125)
   Memory: 0B
   CGroup: /system.slice/bootkube.service

Oct 25 19:34:03 localhost.localdomain systemd[1]: Started Bootstrap a Kubernetes cluster.
Oct 25 19:34:03 localhost.localdomain systemd[1]: Starting Bootstrap a Kubernetes cluster...
Oct 25 19:34:04 localhost.localdomain bootkube.sh[913]: Trying to pull registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest...Failed
Oct 25 19:34:05 localhost.localdomain bootkube.sh[913]: unable to pull registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest: unable to pull image: Error determining manifest MIME type for docker://registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest: pinging docker...
Oct 25 19:34:05 localhost.localdomain systemd[1]: bootkube.service: main process exited, code=exited, status=125/n/a
Hint: Some lines were ellipsized, use -l to show in full.
[core@ip-10-0-6-79 ~]$ ps aux | grep 913
core     16482  0.0  0.0  10688   972 pts/0    S+   21:07   0:00 grep --color=auto 913
[core@ip-10-0-6-79 ~]$ journalctl -u bootkube
-- Logs begin at Thu 2018-10-25 19:33:29 UTC, end at Thu 2018-10-25 21:08:07 UTC. --
Oct 25 19:34:03 localhost.localdomain systemd[1]: Started Bootstrap a Kubernetes cluster.
Oct 25 19:34:03 localhost.localdomain systemd[1]: Starting Bootstrap a Kubernetes cluster...
Oct 25 19:34:04 localhost.localdomain bootkube.sh[913]: Trying to pull registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest...Failed
Oct 25 19:34:05 localhost.localdomain bootkube.sh[913]: unable to pull registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest: unable to pull image: Error determining manifest MIME type for docker://registry.svc.ci.openshift.org/ci-op-nvfgsy47/release:latest: pinging docker re
Oct 25 19:34:05 localhost.localdomain systemd[1]: bootkube.service: main process exited, code=exited, status=125/n/a

[core@ip-10-0-6-79 ~]$ systemctl cat bootkube
# /etc/systemd/system/bootkube.service

[Unit]
Description=Bootstrap a Kubernetes cluster
Wants=kubelet.service
After=kubelet.service

[Service]
WorkingDirectory=/opt/tectonic
RemainAfterExit=true
ExecStart=/opt/tectonic/bootkube.sh

Restart=on-failure
RestartSec=5s
```
The bootkube service failed die to dns blip and was not restarted

# What you expected to happen?

The bootkube service should have started.

**Comment by jianzhangbjz** [comment]
I also encounter this issue by using **libvirt**. Here are my steps:
1) create a new cluster via the openshift-installer.
```console
[jzhang@dhcp-140-18 installer]$ openshift-install create cluster --dir 1107
? Image [? for help] (https://releases-rhcos.svc.ci.openshift.org/storage/releases/maipo/47.77/redhat-coreos-maipo-47.77-qemu.qcow? Image https://releases-rhcos.svc.ci.openshift.org/storage/releases/maipo/47.77/redhat-coreos-maipo-47.77-qemu.qcow2
INFO Fetching OS image...                         
INFO Using Terraform to create cluster...         
[jzhang@dhcp-140-18 installer]$ sudo virsh list
setlocale: No such file or directory
 Id    Name                           State
----------------------------------------------------
 27    bootstrap                      running
 28    master0                        running
```
2)Created it failed, access the `bootstrap` and check the logs:
```console
[jzhang@dhcp-140-18 installer]$ ssh core@test2-bootstrap.tt.testing
[core@test2-bootstrap ~]$ journalctl bootkube -u
journalctl: option requires an argument -- 'u'
[core@test2-bootstrap ~]$ journalctl -u bootkube   
-- Logs begin at Wed 2018-11-07 06:12:47 UTC, end at Wed 2018-11-07 06:41:06 UTC. --
Nov 07 06:13:02 test2-bootstrap systemd[1]: Started Bootstrap a Kubernetes cluster.
Nov 07 06:13:02 test2-bootstrap systemd[1]: Starting Bootstrap a Kubernetes cluster...
Nov 07 06:13:22 test2-bootstrap bootkube.sh[998]: Trying to pull registry.svc.ci.openshift.org/openshift/origin-release:v4.0...Failed
Nov 07 06:13:22 test2-bootstrap bootkube.sh[998]: unable to pull registry.svc.ci.openshift.org/openshift/origin-release:v4.0: unable to pull image: Error determining manife
Nov 07 06:13:22 test2-bootstrap systemd[1]: bootkube.service: main process exited, code=exited, status=125/n/a
```
So, how to solve it?

**Comment by wking** [comment]
> ```
> $ openshift-install create cluster --dir 1107
> ? Image [? for help] (https://releases-rhcos.svc.ci.openshift.org/storage/releases/maipo/47.77/redhat-coreos-maipo-47.77-qemu.qcow? Image https://releases-rhcos.svc.ci.openshift.org/storage/releases/maipo/47.77/redhat-coreos-maipo-47.77-qemu.qcow2
> INFO Fetching OS image...
> INFO Using Terraform to create cluster...
> ```

What version is this?  Looks like it's missing #579.  Retry with a fresh build from master?

**Comment by wking** [comment]
Fixed by #604?

**Comment by jianzhangbjz** [comment]
Yes. I rebuild it with the latest master branch. Seems like this problem had been fixed. Thanks!
```console
[jzhang@dhcp-140-18 ~]$ openshift-install version
openshift-install v0.3.0-155-g25ceecc296bd020219967ae1258180df01acff0f
Terraform v0.11.8

Your version of Terraform is out of date! The latest version
is 0.11.10. You can update by downloading from www.terraform.io/downloads.html
```
